pycbc/waveform/premerger.py

```diff
@@ -0,0 +1,33 @@
+""" Waveform approximants for the pre-merger detection of gravitational waves
+"""
+import logging
+
+
+def premerger_taylorf2(**p):
+    """ Generate time-shifted TaylorF2"""
+    from pycbc.waveform import get_fd_waveform
+    from pycbc.waveform.spa_tmplt import spa_length_in_time
+
+    p.pop('approximant')
+    hp, hc = get_fd_waveform(approximant="TaylorF2", **p)
+
+    removed = spa_length_in_time(mass1=p['mass1'],
+                                 mass2=p['mass2'],
+                                 f_lower=p['f_final'],
+                                 phase_order=-1)
```

Comment:
Probably not an issue for the current application, but do we have a function similar to `spa_length_in_time()` that includes PN spin effects in PyCBC?

Reply:
Hmm, I don't think so.